prov/rxd/src/rxd.h

```diff
@@ -191,6 +191,11 @@ struct rxd_ep {
 	uint32_t tx_flags;
 	uint32_t rx_flags;
 
+	size_t tx_msg_avail;
+	size_t rx_msg_avail;
+	size_t tx_rma_avail;
+	size_t rx_rma_avail;
```

Comment:
I think these end up corresponding to buffer pool max_cnt attributes.  Can you just set the max_cnt and let the buffer pool fail allocations to support these limits instead?

Reply:
Right now there are only 2 pools: tx and rx. But we should distinguish between msg and rma ops, right? In other words, if the user posts rx_size receive buffers, it should still be able to process incoming RMA ops without having to wait until all of those buffers are used. Otherwise, I could separate the pools into 4 pools, eek.

Reply:
discussed off line